pygac_fdr/metadata.py

```diff
@@ -138,47 +167,69 @@ def get_metadata(self, filenames):
+            eq_cross_lons, eq_cross_times = self._get_equator_crossings(ds)
+            metadata = FileMetadata(
+                platform=ds.attrs['platform'].split('>')[-1].strip(),
+                start_time=ds['acq_time'].values[0],
+                end_time=ds['acq_time'].values[-1],
+                along_track=ds.dims['y'],
+                filename=filename,
+                orbit_number_start=ds.attrs['orbit_number_start'],
+                orbit_number_end=ds.attrs['orbit_number_end'],
+                equator_crossing_longitude_1=eq_cross_lons[0],
+                equator_crossing_time_1=eq_cross_times[0],
+                equator_crossing_longitude_2=eq_cross_lons[0],
+                equator_crossing_time_2=eq_cross_times[1],
+                midnight_line=midnight_line,
+                overlap_free_start=np.nan,
+                overlap_free_end=np.nan,
+                global_quality_flag=QualityFlags.OK
+            )
+        return metadata
+
     def _collect_metadata(self, filenames):
         """Collect metadata from the given level 1c files."""
-        records = []
-        for filename in filenames:
-            LOG.debug('Collecting metadata from {}'.format(filename))
-            with xr.open_dataset(filename) as ds:
-                midnight_line = np.float64(self._get_midnight_line(ds['acq_time']))
-                eq_cross_lons, eq_cross_times = self._get_equator_crossings(ds)
-                rec = {'platform':  ds.attrs['platform'].split('>')[-1].strip(),
-                       'start_time': ds['acq_time'].values[0],
-                       'end_time': ds['acq_time'].values[-1],
-                       'along_track': ds.dims['y'],
-                       'filename': filename,
-                       'orbit_number_start': ds.attrs['orbit_number_start'],
-                       'orbit_number_end': ds.attrs['orbit_number_end'],
-                       'equator_crossing_longitude_1': eq_cross_lons[0],
-                       'equator_crossing_time_1': eq_cross_times[0],
-                       'equator_crossing_longitude_2': eq_cross_lons[1],
-                       'equator_crossing_time_2': eq_cross_times[1],
-                       'midnight_line': midnight_line,
-                       'overlap_free_start': np.nan,
-                       'overlap_free_end': np.nan,
-                       'global_quality_flag': QualityFlags.OK}
-                records.append(rec)
-        return records
+        if os.path.isfile(CHECKPOINT_DB):
+            LOG.info("Restarting from checkpoint.")
+        db_url = 'sqlite:///{0}'.format(CHECKPOINT_DB)
+        engine = sqlalchemy.create_engine(db_url)
+        Base.metadata.create_all(engine)
+        # open session
+        Session = sessionmaker(bind=engine)
+        session = Session()
+        # get set of processed files in case of a restart
+        wanted = set(filenames)
+        done = set(
+            filename for filename, in session.query(FileMetadata.filename)
```

Comment:
Is that `,` intended?

Reply:
Unfortunately yes, because the query returns tuples (of one item) and therefore I need to unpack them.

Reply:
Have not yet seen a better way to get directly a list for a 1 column query.

Reply:
Ok I see. What about `for filename, _ in ...` ?

Reply:
Another idea:
```python
import itertools as it
done = set(it.chain(*session.query(FileMetadata.filename)))
```
Which is another way to flatten the list of one element tuples.

Reply:
Or another way without any import:
```python
done = set(next(zip(*session.query(FileMetadata.filename))))
```